src/layers.cpp

```diff
@@ -10,8 +10,9 @@ Key(*Layer_::getKey)(uint8_t layer, byte row, byte col) = Layer.getKeyFromPROGME
 
 // The total number of defined layers in the firmware sketch keymaps[]
 // array. If the keymap wasn't defined using CREATE_KEYMAP() in the
-// sketch file, layer_count gets the default value of zero.
-uint8_t layer_count __attribute__((weak)) = 0;
+// sketch file, layer_count gets the default value of 32 (the highest
+// possible layer; LayerState has only 32 bits).
+uint8_t layer_count __attribute__((weak)) = 32;
```

Comment:
Rather than hardcoding the magic constant of 32, I think it makes sense to define a compiler macro of MAX_LAYERS or somesuch, somewhere and to use that.

Reply:
Agreed. I started to write:
```
sizeof(LayerState) * 8 - 1
```
…but then thought that wasn't really helping. I like defining it as a constant (preprocessor macro).

Reply:
Defining the constant that way, however, is _great_. It explains exactly what's going on. Suddenly it's no longer a magic constant.